02/app.js

```diff
@@ -1,7 +1,21 @@
-
 /* rozwiązanie z pętlą for */
-const x = 4;
+const x = Number(prompt("Podaj liczbę!"));
+validateNumer(x);
+const isNumberValid = x > 0 && x < 10;
+if (!isNumberValid) throw new Error("given number not in range from 1 to 9");
 
+for (let i = 1; i < 10; i++) {
+  console.log(`${x} x ${i} = ${x * i}`);
+}
 
+/* rozwiązanie z pętlą while  */
+// const a = Number(prompt("Podaj podstawę!"));
+// const n = Number(prompt("Podaj wykładnik"));
+// validateNumer(a);
+// validateNumer(n);
+// alert(`Wynik potęgowania ${a} i ${n} to ${Math.pow(3, 4)}`);
 
-/* rozwiązanie z pętlą while  */
\ No newline at end of file
+function validateNumer(number) {
+  const isNumberType = typeof number === "number" && !isNaN(number);
+  if (!isNumberType) throw new Error("parameter is not a number type");
```

Comment:
Rozwiązanie z rzucaniem błędu w globalnej przestrzeni nie jest najlepszym pomysłem bo nie ma jak tego przechwycić - wystarczy prostu `if` - dopiero przy rozbudowanej strukturze nabiera to większego sensu ;)